core/node/node_framework/src/implementations/layers/query_eth_client.rs

```diff
@@ -0,0 +1,34 @@
+use std::sync::Arc;
+
+use anyhow::Context;
+use zksync_eth_client::clients::QueryClient;
+
+use crate::{
+    implementations::resources::eth_interface::EthInterfaceResource,
+    service::ServiceContext,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+#[derive(Debug)]
+pub struct QueryEthClientLayer {
+    web3_url: String,
+}
+
+impl QueryEthClientLayer {
+    pub fn new(web3_url: String) -> Self {
```

Comment:
It's a bit inconsistent with other layers that accept whole config structures, but it didn't feel right to depend on the whole config for the sake of just one variable.

Reply:
should we separate configs then?

Reply:
Probably we should, but it's out of scope for this PR and a part of the larger topic of reforming the configuration system for the node.